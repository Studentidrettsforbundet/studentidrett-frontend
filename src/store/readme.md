# Redux

This folder contains all redux files.

Add this extension to your browser: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd

## Description:

store.ts is the file that creates the redux store, as well as the combined reducer. All reducers needs to be added to the combined reducer, and to the combinedState

This folder contains a folder for each page of our application that needs redux.
The folders contains reducers, actions, action crators, types, interfaces, etc. related to pages.

## Example of folder structure

We have a page called "search". That page needs a reducer, some actions, action creators, and some interfaces. Below is an example of the folder structure

----/store
--------store.ts
--------/pages
-----------/search
-----------------searchReducer.ts
-----------------searchActions.ts
-----------------searchReducer.ts
