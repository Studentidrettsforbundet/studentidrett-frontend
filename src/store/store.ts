import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { cityInitialState, cityReducer, cityState } from './pages/city/cityReducer';
import { clubInitialState, clubReducer, clubDetailReducer, clubState } from './pages/club/clubReducer';
import { groupInitialState, groupReducer, groupDetailReducer, groupState } from './pages/group/groupReducer';
import { regionInitialState, regionReducer, regionState } from './pages/region/regionReducer';
import { sportInitialState, sportReducer, sportDetailReducer, sportState } from './pages/sport/sportReducer';
import { teamInitialState, teamReducer, teamDetailReducer, teamState } from './pages/team/teamReducer';
import { searchBarInitialState, searchBarReducer, searchBarState } from './searchBar/searchBarReducer';
import { thunkInitialState, thunkReducer, thunkState } from './thunks/thunkReducer';
import { searchInitialState, searchReducer, searchState } from './pages/search/searchReducer';
import {
    questionInitialState,
    questionnaireReducer,
    questionnaireState,
} from './pages/questionnaire/questionnaireReducer';

declare global {
    interface Window {
        __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

//TODO: add reducers interfaces
export interface combinedStateInterface {
    region: regionState;
    city: cityState;
    sport: sportState;
    sport_detail: sportState;
    club: clubState;
    club_detail: clubState;
    team: teamState;
    team_detail: teamState;
    group: groupState;
    group_detail: groupState;
    search_results: searchState;
    thunk: thunkState;
    searchBar: searchBarState;
    questionnaire: questionnaireState;
}

// TODO: add all pages state here
export const combinedState = {
    region: regionInitialState,
    city: cityInitialState,
    sport: sportInitialState,
    sport_detail: sportInitialState,
    club: clubInitialState,
    club_detail: clubInitialState,
    team: teamInitialState,
    team_detail: teamInitialState,
    group: groupInitialState,
    group_detail: groupInitialState,
    search_results: searchInitialState,
    thunk: thunkInitialState,
    searchBar: searchBarInitialState,
    questionnaire: questionInitialState,
};

// create store
const store = createStore(
    combineReducers({
        region: regionReducer,
        city: cityReducer,
        sport: sportReducer,
        sport_detail: sportDetailReducer,
        club: clubReducer,
        club_detail: clubDetailReducer,
        team: teamReducer,
        team_detail: teamDetailReducer,
        group: groupReducer,
        group_detail: groupDetailReducer,
        search_results: searchReducer,
        thunk: thunkReducer,
        searchBar: searchBarReducer,
        questionnaire: questionnaireReducer,
    }),
    combinedState,
    composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
