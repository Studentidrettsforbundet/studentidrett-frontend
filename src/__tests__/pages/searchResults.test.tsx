import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import store from '../../store/store';
import 'mutationobserver-shim';
import SearchResults from '../../pages/searchResults';

describe('regionPage', () => {
    global.MutationObserver = window.MutationObserver;
    test('renders correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <SearchResults />
                </MemoryRouter>
            </Provider>,
        );
        expect(tree).toMatchSnapshot();
    });
});
