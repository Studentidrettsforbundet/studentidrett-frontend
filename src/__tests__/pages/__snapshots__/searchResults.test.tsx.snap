// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regionPage renders correctly 1`] = `
Array [
  <div>
    <div
      className="f3518ct card"
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <input
          className="form-control"
          name="searchString"
          onChange={[Function]}
          placeholder="Søk..."
          value=""
        />
        <a
          href="/search/?q="
          onClick={[Function]}
        >
          <button
            className="searchBar_button btn btn-primary"
            disabled={true}
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#808080",
                "color": "#808080",
              }
            }
            type="submit"
          >
            Søk
          </button>
        </a>
      </div>
      <p
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
         Søkefiltre: 
      </p>
      <div
        className="f1v27uv7"
      >
        <div
          className="faz2bcf"
        >
          <button
            className="f3b1glp btn btn-secondary outlined"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "color": "black",
              }
            }
            type="button"
            value="cities"
          >
            By
          </button>
        </div>
        <div
          className="faz2bcf"
        >
          <button
            className="f3b1glp btn btn-secondary outlined"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "color": "black",
              }
            }
            type="button"
            value="clubs"
          >
            Klubb
          </button>
        </div>
        <div
          className="faz2bcf"
        >
          <button
            className="f3b1glp btn btn-secondary outlined"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "color": "black",
              }
            }
            type="button"
            value="groups"
          >
            Gruppe
          </button>
        </div>
        <div
          className="faz2bcf"
        >
          <button
            className="f3b1glp btn btn-secondary outlined"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "color": "black",
              }
            }
            type="button"
            value="teams"
          >
            Lag
          </button>
        </div>
        <div
          className="faz2bcf"
        >
          <button
            className="f3b1glp btn btn-secondary outlined"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "color": "black",
              }
            }
            type="button"
            value="sports"
          >
            Idrett
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div />
  </div>,
]
`;
